furms-core/src/main/java/io/imunity/furms/core/sites/SiteServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020 Bixbit s.c. All rights reserved.
+ *  See LICENSE file for licensing information.
+ */
+
+package io.imunity.furms.core.sites;
+
+import io.imunity.furms.api.sites.SiteService;
+import io.imunity.furms.domain.sites.Site;
+import io.imunity.furms.spi.sites.SiteRepository;
+import io.imunity.furms.spi.sites.SiteWebClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+import java.util.Set;
+
+@Service
+class SiteServiceImpl implements SiteService {
+
+	private static final Logger LOG = LoggerFactory.getLogger(SiteServiceImpl.class);
+
+	private final SiteRepository siteRepository;
+	private final SiteServiceValidator validator;
+	private final SiteWebClient webClient;
+
+	SiteServiceImpl(SiteRepository siteRepository,
+	                SiteServiceValidator validator,
+	                SiteWebClient webClient) {
+		this.siteRepository = siteRepository;
+		this.validator = validator;
+		this.webClient = webClient;
+	}
+
+	@Override
+	public Optional<Site> findById(String id) {
```

Comment:
@SooLiD  I'll merge it to dev branch, however please add authZ for site service methods, this is must have. Take a look at communities service for reference.